### ISSUE TITLE
Performance Issues Overall.

### ISSUE THREAD

**Issue by StoneLegion** [bdew-minecraft/generators#32]
So we been having lag for weeks and Opis was not really the best in it but we used that profile mod. 

So I typed /profile e after while when the server is lagging. It seems to not be right away but an overtime leak that happens. This is some the results we had with maybe 12+ hours uptime on it.

All Entities of Type | Time/Tick |      %
  TileSteamTurbineController |    24.536 | 62.362

Single Entity | Time/Tick |     %
TileSteamTurbineController 94,27,-472:0 |    12.684 | 2.346
TileSteamTurbineController 94,27,-474:0 |    12.578 | 2.327
TileSteamTurbineController 94,27,-480:0 |    12.383 | 2.291
TileSteamTurbineController 94,27,-478:0 |    12.309 | 2.277
TileSteamTurbineController 94,27,-476:0 |    12.273 | 2.270

Worst part we could not even find any steam turbines active they seem to have no real progress prob out of the materials etc. This another players base have not seen him in days and don't know much of the mod yet. 

Here is screenshots of his setup a little fing too much as you can see:
Main View:
http://i.imgur.com/zojuqwl.png
Future Additions?:
http://i.imgur.com/fmNtapm.png
Another small setup?
http://i.imgur.com/ThIblUq.png
This block here seems to be quite taxing in opis:
http://i.imgur.com/puAlRZN.png

In conclusion I have a feeling there might be a CPU leak overtime. I doubt I can provide more info on this but I hope this might help and shine some light.

**Comment by bdew** [comment]
First of all... that's a lot of turbines :P I'd imagine a similar amount of power generation from other mods would cause some lag as well...

What version is this on? Judging by the textures it looks like one of the recent alpha builds, and i've just re-written a good chunk of the steam turbine code. There might still be some bugs and/or performance issues in it that need ironing out.

Also "CPU leak" doesn't make much sense to me, the code that runs there is the same thing for every tick. It might be a memory leak somewhere that causes more GC activity and exacerbates performance issues.

**Comment by StoneLegion** [comment]
Okay well for science fun I will go ahead and post a profile now and post another in a few hours. See what happens. Strange thing is it's not even running the chunks should not even be loaded but I guess something might be keeping it loaded. 

[10:26:59] [TickProfiler/INFO] [TickProfiler]:
TPS: 20.032665
                                        Single Entity | Time/Tick |     %
                     entity.EntityGoblin -73,75,-62:0 |     0.091 | 2.042
                TileExchangerController 102,26,-484:0 |     0.062 | 1.378
          TileEntityTransferNodeInventory 87,6,-504:0 |     0.051 | 1.152
BlockWitchesOven$TileEntityWitchesOven 1615,68,1346:0 |     0.043 | 0.972
            TileEntityTransferNodeLiquid 108,7,-487:0 |     0.039 | 0.867
     Chunk | Time/Tick |     %
 0: 5, -30 |     0.411 | 9.210
 0: 6, -31 |     0.383 | 8.567
0: 100, 82 |     0.382 | 8.559
 0: 5, -29 |     0.296 | 6.623
 0: 5, -31 |     0.273 | 6.109
      All Entities of Type | Time/Tick |      %
       turbine.TileTurbine |     0.561 | 12.569
         TileConduitBundle |     0.426 |  9.550
TileSteamTurbineController |     0.365 |  8.166
     passive.EntityChicken |     0.143 |  3.202
         TileTankSteelWall |     0.112 |  2.519
         Average Entity of Type | Time/tick | Calls
            entity.EntityGoblin |     0.091 |   601
TileEntityTransferNodeInventory |     0.051 |   601
        TileExchangerController |     0.043 |  1202
         TileEntityChargepadMFE |     0.038 |   601
           TileSyngasController |     0.037 |   601

**Comment by StoneLegion** [comment]
So it's been about 8 hours or so. I wanted to share another report on the performance. It seems to get worse I have no clue why the odd thing unless chunks naturally loading and that happens of course vanilla/other mods.

Here is what we have:
[18:42:23] [TickProfiler/INFO] [TickProfiler]:
TPS: 16.032799
                          Single Entity | Time/Tick |     %
TileSteamTurbineController 92,27,-450:0 |     2.308 | 2.288
TileSteamTurbineController 92,27,-460:0 |     2.300 | 2.280
TileSteamTurbineController 92,27,-462:0 |     2.292 | 2.272
TileSteamTurbineController 92,27,-456:0 |     2.291 | 2.271
TileSteamTurbineController 92,27,-452:0 |     2.291 | 2.271
    Chunk | Time/Tick |      %
0: 5, -30 |     5.239 | 23.791
0: 5, -29 |     4.259 | 19.339
0: 5, -31 |     2.320 | 10.534
0: 36, 36 |     1.217 |  5.527
0: 6, -31 |     1.156 |  5.249
        All Entities of Type | Time/Tick |      %
  TileSteamTurbineController |    12.517 | 56.839
TileEntityElectricFloodlight |     1.227 |  5.574
         turbine.TileTurbine |     0.817 |  3.708
     TileExchangerController |     0.783 |  3.553
           TileConduitBundle |     0.706 |  3.206
      Average Entity of Type | Time/tick | Calls
     TileExchangerController |     0.261 |  1442
  TileSteamTurbineController |     0.213 | 28297
       player.EntityPlayerMP |     0.088 |   481
          TileInfusionMatrix |     0.066 |   481
TileEntityElectricFloodlight |     0.049 | 12016

**Comment by bdew** [comment]
I'm closing this issue as part of a spring cleaning of my issue trackers from old stuff.

If you believe it's a bug that is still relevant in Minecraft 1.12 - Feel free to reopen it.
